presto-main/src/main/java/com/facebook/presto/sql/gen/JoinCompiler.java

```diff
@@ -616,22 +672,68 @@ private static void generatePositionEqualsRowWithPageMethod(
 
         Variable thisVariable = positionEqualsRowMethod.getThis();
         BytecodeBlock body = positionEqualsRowMethod.getBody();
-
         for (int index = 0; index < joinChannelTypes.size(); index++) {
             BytecodeExpression type = constantType(callSiteBinder, joinChannelTypes.get(index));
-
             BytecodeExpression leftBlock = thisVariable
                     .getField(joinChannelFields.get(index))
                     .invoke("get", Object.class, leftBlockIndex)
                     .cast(Block.class);
 
             BytecodeExpression rightBlock = page.invoke("getBlock", Block.class, rightChannels.getElement(index));
-
             body.append(new IfStatement()
                     .condition(typeEquals(type, leftBlock, leftBlockPosition, rightBlock, rightPosition))
                     .ifFalse(constantFalse().ret()));
         }
+        body.append(constantTrue().ret());
+    }
+
+    private void generatePositionNotDistinctFromRowWithPageMethod(
+            ClassDefinition classDefinition,
+            CallSiteBinder callSiteBinder,
+            List<Type> joinChannelTypes,
+            List<FieldDefinition> joinChannelFields)
+    {
+        if (groupByUsesEqualTo) {
+            generatePositionEqualsRowWithPageMethod(classDefinition, callSiteBinder, joinChannelTypes, joinChannelFields, true);
```

Comment:
When it's in the legacy groupByUsesEqualTo, the `positionNotDistinctFromRow` will be generated as the same as positionEqualsRow.

However, the current implementation will directly delegate to `generatePositionEqualsRowWithPageMethod` method, with a flag indicates the method name. So, the `generatePositionEqualsRowWithPageMethod` can generate a method called `positionNotDistinctFromRow`, despite its name says it's generating `positionEqualsRow`. This is confusing.

Can we just make the generated `positionNotDistinctFromRow` delegate to call `positionEqualsRow` in the legacy mode?

Reply:
Makes sense, thanks!